test/test_state_update.py

```diff
@@ -63,7 +66,7 @@ def test_deployed_contracts():
     deployed_contracts = state_update["state_diff"]["deployed_contracts"]
 
     assert len(deployed_contracts) == 1
-    assert deployed_contracts[0]["address"] == contract_address
+    assert int(deployed_contracts[0]["address"], 16) == int(contract_address, 16)
```

Comment:
Here and in other assertions introduced by this PR, can you use the `assert_equal` function from util.py? It should give a more informative output on failed assertion